### PR TITLE
Added method to parse user_and_roles string

### PR DESCRIPTION
**Pull request by skkosuri-amzn**
*Issue #, if available:*
Added method to parse user_and_roles string

*Description of changes:*
Added method to parse user_and_roles string

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/java/com/amazon/opendistroforelasticsearch/commons/authuser/User.java [hunk 1]
```diff
@@ -134,6 +135,34 @@ public static User parse(XContentParser parser) throws IOException {
         return new User(name, backendRoles, roles, customAttNames);
     }
 
+    /**
+     * User String format must be pipe separated as : user_name|backendrole1,backendrole2|roles1,role2
```
Comment (ylwu-amzn) [inline src/main/java/com/amazon/opendistroforelasticsearch/commons/authuser/User.java:139]: We should not allow user input "|" in user/role name. Do we have such checking ?

File: src/main/java/com/amazon/opendistroforelasticsearch/commons/authuser/User.java [hunk 2]
```diff
@@ -134,6 +135,34 @@ public static User parse(XContentParser parser) throws IOException {
...
+        List<String> backendRoles = new ArrayList<>();
+        List<String> roles = new ArrayList<>();
+
+        if ((strs.length > 1) && !Strings.isNullOrEmpty(strs[1])) {
+            backendRoles.addAll(Arrays.asList(strs[1].split(",")));
+        }
+        if ((strs.length > 2) && !Strings.isNullOrEmpty(strs[2])) {
+            roles.addAll(Arrays.asList(strs[2].split(",")));
+        }
+        return new User(userName, backendRoles, roles, Arrays.asList());
```
Comment (ylwu-amzn) [inline src/main/java/com/amazon/opendistroforelasticsearch/commons/authuser/User.java:163]: Is it by design to not support custom attribute names?

Comment (skkosuri-amzn) [reply src/main/java/com/amazon/opendistroforelasticsearch/commons/authuser/User.java:163]: Yes. And also we want to keep injected string light weight. Custom attributes could make the string large.

File: src/main/java/com/amazon/opendistroforelasticsearch/commons/authuser/User.java (hunk 1 above)
Comment (skkosuri-amzn) [reply src/main/java/com/amazon/opendistroforelasticsearch/commons/authuser/User.java:139]: Security plugin has those checks, we use similar format for UserInjection and RolesInjection
https://github.com/opendistro-for-elasticsearch/security/blob/6304e7875ba7dc24ffcb0fe89a875a56cb6cc853/src/main/java/com/amazon/opendistroforelasticsearch/security/auth/RolesInjector.java#L35
https://github.com/opendistro-for-elasticsearch/security/blob/b69e821a360a774ec2e94dc01390cd61521562d2/src/main/java/com/amazon/opendistroforelasticsearch/security/auth/UserInjector.java#L84

Comment (skkosuri-amzn) [reply src/main/java/com/amazon/opendistroforelasticsearch/commons/authuser/User.java:139]: Related PR: https://github.com/opendistro-for-elasticsearch/security/pull/798

**Review by lezzago — APPROVED** [review]

**Review by ylwu-amzn — APPROVED** [review]

**Pull request merged**